docs-src/guidelines.md

```diff
@@ -83,3 +83,20 @@ We expect there will be varying levels of conformance among the
 different providers in the early days of this API. Users can use the
 results of the conformance tests to understand areas where there may
 be differences in behavior from the spec.
+
+### Conformance Tests
+
+Conformance tests are actively being developed to ensure that implementations of
+this API are conformant with the spec. Use `make conformance` to run these tests
+with the Kubernetes cluster you are currently connected to. These tests require
+the `CONTROLLER_NAME` environment variable to be set. This will be used to set
+the `controller` field on GatewayClass resources created for the conformance
+tests.
```

Comment:
Do we expect controller name to be hardcoded in implementations?
If yes, then we have to do this.
An alternate approach would be if we hard-code the controller name in this project and implementations have a configurable parameters (environment variable) that can be set when running conformance (and also used by end users).

Reply:
On a second look, another approach comes to mind:
- Hardcode the GatewayClass name in Gateway resources used in tests
- `GatewayClass` resource is supplied as part of the implementation manifests, this has the added advantage of an implementation being able to use `gatewayclass.spec.parametersRef` field.


I'd like to avoid templating YAML as much as possible. It is harmless in this patch but my fear is that it will get out of hand pretty quickly.